### ISSUE TITLE
Inconsistent LIF time-axis handling in `FB_LiDiff_Attention.forward`

### ISSUE THREAD

**Issue by jazzypanda** [KirinZheng/SpiLiFormer#4]
# Inconsistent LIF time-axis handling in `FB_LiDiff_Attention.forward`

The final `proj_lif` in `FB_LiDiff_Attention` is applied **before** the reshape
back to `(T, B, …)` — every other LIF call in the model reshapes first. With
`B > 1` this makes `MultiStepLIFNode` recur over `T·B` "timesteps" with
membrane voltage leaking across batch samples.

## The line

`imagetnet_1k/spiliformer.py:156-157`:

```python
x = x.flatten(0, 1)
x = self.proj_lif(self.proj_bn(self.proj_conv(x))).reshape(T, B, C, W, H)
```

Compare with the same op in `FF_LiDiff_Attention` two classes above:

```python
x = self.proj_bn(self.proj_conv(x.flatten(0, 1))).reshape(T, B, C, H, W)
x = self.proj_lif(x)
```

Same pattern (reshape → LIF) is used in `MLP`, `PatchEmbedInit` (×3), `PatchEmbeddingStage` (×3),
and `FB_LiDiff_Attention.attn_lif` itself.

## Repro

Two copies of the model, both loading `checkpoint_spiliformer_T4_224.pth` with
`strict=True` (478 keys match exactly):

- `m_orig` — released code
- `m_fix` — identical except `FB_LiDiff_Attention` reshapes before `proj_lif`

```text
bs= 1   max_abs(logits) = 0.000    argmax agree = 1.00
bs= 2   max_abs(logits) = 0.677    argmax agree = 0.50
bs= 4   max_abs(logits) = 0.841    argmax agree = 0.00
bs= 8   max_abs(logits) = 1.043    argmax agree = 0.00
bs=32   max_abs(logits) = 1.090    argmax agree = 0.03
```

`B = 1` is bit-identical (because `T·B = T`). Layer-by-layer sweep at `B = 2`
localises the first divergence to `stage3[0].attn`; `q`, `k`, `v`, and
`attn_lif` outputs match — divergence appears exactly at the final projection.

Reverting `m_fix` to the released ordering recovers bit-identity at every
batch size.

## Effect

With `B > 1`, sample `b`'s timestep-`(T-1)` voltage initialises sample `b+1`'s
timestep-`0` integration for the deep-stage `proj_lif`. State is reset between
batches by `functional.reset_net`, so the leakage is intra-batch only and
order-dependent across the batch slots.

Released checkpoint was trained this way; reordering the line changes outputs
non-trivially (above) so it can't be patched in place without breaking
`strict=True` loading.

Full ImageNet val acc with the released checkpoint, single-pass eval, BF16:

| FB proj_lif ordering | Top-1 | Top-5 | throughput |
|----------------------|-------|-------|------------|
| released (LIF on `(T·B, C, N)`) | **84.84%** | 97.33% | 139 img/s |
| consistent (LIF on `(T, B, …)`) | **78.15%** | 93.65% | 241 img/s |
| Δ | **−6.69** | −3.68 | +73% |

So the line isn't just "out of style" — the model's weights have learned to
operate under intra-batch voltage leakage, and dropping the leakage at
inference costs ~6.7 Top-1 points. The throughput gap is also real: the
released ordering makes the deep `proj_lif` run a sequence of length `T·B`
instead of `T`.

## Suggested resolution

Pick one — happy to PR whichever you prefer:

1. Add a comment marking the ordering as load-bearing for the released checkpoint.
2. Gate it behind a flag (`legacy_proj_lif=True`) so future training can use the
   consistent ordering while existing weights still load.
3. Retrain with the consistent ordering and publish a new checkpoint.

## Reproduction snippet

```python
import torch
from spiliformer import SpiLiFormer_10_768                    # released
from spiliformer_kit.model import SpiLiFormer_10_768 as Fixed # reshape-before-LIF variant

m_orig = SpiLiFormer_10_768(T=4).cuda().eval()
m_fix  = Fixed(T=4).cuda().eval()

state = torch.load("checkpoint_spiliformer_T4_224.pth",
                   map_location="cpu", weights_only=False)["model"]
m_orig.load_state_dict(state, strict=True)
m_fix.load_state_dict(state,  strict=True)

torch.manual_seed(0)
for bs in [1, 2, 4, 32]:
    x = torch.randn(bs, 3, 224, 224, device="cuda")
    with torch.no_grad():
        o_orig = m_orig(x)[0]
        o_fix  = m_fix(x)[0]
    print(f"bs={bs:>2}  max_abs={(o_orig-o_fix).abs().max():.3f}  "
          f"argmax_agree={(o_orig.argmax(1)==o_fix.argmax(1)).float().mean():.2f}")
```

**Comment by KirinZheng** [comment]
Thank you very much for your question and for your careful attention to our work. I truly appreciate the detailed report and reproduction.

I am currently occupied with several other tasks, so I may not be able to provide a detailed response immediately. It may take me a bit longer to get back to you, as I have quite a lot on my plate at the moment. However, I will carefully look into this issue and provide a more thorough reply later.

Thanks again for your valuable feedback.
